enumerable.rb

```diff
@@ -0,0 +1,349 @@
+          if element == word
+            k += 1
+          end
+        end
+        k > 0 ? true : false
+
+      elsif word == Numeric
+        for element in self
+          if element.class <= word
+            j += 1
+          end
+        end
+
+        j > 0 and j <= self.length ? true : false
+
+      else
+        k = 0
+        for element in self do
+          if element == false or element == nil
+            k += 1
+          end
+        end
+        k != self.size ? true : false
+
+      end
+
+    else
+      i = 0
+      j = 0
+      while i < self.size
+        yield self[i]
+        i += 1
+      end
+
+      for element in self
+        if yield element
+          j += 1
+        end
+      end
+      j > 0 and j <= self.size ? true : false
+    end
+  end
+
+  def my_none?(word = nil)
+    unless block_given?
+      i = 0
+      j = 0
+      while i < self.size
+        self[i]
+        i += 1
+      end
+      if self.size == 0
+        true
+      elsif word.class == Regexp
+
+        for element in self
+          if element.match(word)
+            j += 1
+          end
+        end
+        j == 0 ? true : false
+
+      elsif word != nil
+        if word == Numeric
+          for element in self
+            if element.class <= word
+              j += 1
+            end
+          end
+
+          j != 0 ? false : true
+        else
+          i = 0
+          k = 0
+          while i < self.size
+            if self[i] == word
+              k += 1
+            end
+            i += 1
+          end
+          k == 0 ? true : false
+      end
+
+      else
+        j = 0
+        self.my_each do |e|
+          if e == true
+            j += 1
+          end
+        end
+        j == 0 ? true : false
+      end
+    else
+      i = 0
+      while i < self.size
+        yield self[i]
+        i += 1
+      end
+      j = 0
+      self.my_each do |element|
+        if yield element
+          j += 1
+        end
+      end
+      j == 0 ? true : false
+
+    end
+  end
+
+  def my_count(val = nil)
+    unless block_given?
+
+      if val.class != NilClass
+        i = 0
+
+        for el in self do
+          if el == val
+            i += 1
+          end
+        end
+        i
+      else
+        self.my_each do |count|
+          count += 1
+        end
+        count
+      end
+    else
+      j = 0
+      self.my_each do |element|
+        if yield element
+          j += 1
+        end
+      end
+      j
+    end
+  end
+
+  def my_map
+    unless block_given?
+      to_enum(__method__)
+    else
+      arr = []
+      for element in self do
+        current = yield element
+        arr << current
+      end
+      arr
+    end
+  end
+
+  def my_inject(acc = nil)
+    arr = []
+    i = 0
+    for element in self do
+      arr << element
+    end
+
+    unless block_given?
+
+      if acc.class == Symbol
+        case acc
+        when :+
+          result = 0
+        when :-
+          result = 0
```

Comment:
* [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
5..50.my_inject(2, :*) != 5..50.inject(2, :*)
```